src/dali_executor/dali_pipeline.h

```diff
@@ -70,25 +68,18 @@ class DaliPipeline {
   ~DaliPipeline() {
     ReleasePipeline();
     ReleaseStream();
-    bool is_last_instance = --instance_counter_ == 0;
-    if (is_last_instance && release_buffers_on_delete_) {
```

Comment:
The buggy behaviour is here.

We don't check if the pipeline handle is not null, so we decrease the instance count even in desctructors of null pipelines. 
This, together with the fact that the DaliPipeline object was movable resulted in such a scenario:
```c++
// [instance_counter == 0]
DaliPipeline p(..) // constructor, instance_counter++
auto p2 = std::move(p);

// p2 destructor, instance_counter--
// p destructor, instance_counter--
// [instance_counter == -1]
```